### PR TITLE
feat: change template and examples [MONET-1715]

### PR DESCRIPTION
**Pull request by cemreyuksel**
## Purpose

This PR changes the delivery function templates and examples to function template. 
This PR is a prerequisite for [changing the CLI commands on `create-contentful-app` repository](https://github.com/contentful/create-contentful-app/pull/1844/files) to pass.

### CODE REVIEW

File: examples/function-templates/javascript/contentful-app-manifest.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "functions": [
+    {
+      "id": "example",
+      "name": "Example Function",
+      "description": "This is an example to help you learn how functions work.",
+      "path": "functions/example.js",
+      "entryFile": "functions/example.js",
+      "allowNetworks": []
```
Comment (ghepting) [inline examples/function-templates/javascript/contentful-app-manifest.json:9]: The accepts is missing in both manifest examples

File: examples/function-templates/typescript/contentful-app-manifest.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "functions": [
+    {
+      "id": "example",
+      "name": "Example Function",
+      "description": "This is an example to help you learn how functions work.",
+      "path": "functions/example.js",
+      "entryFile": "functions/example.ts",
+      "allowNetworks": []
```
Comment (ghepting) [inline examples/function-templates/typescript/contentful-app-manifest.json:9]: Accepts?

Comment (cemreyuksel) [reply examples/function-templates/typescript/contentful-app-manifest.json:9]: This just for renaming. That change will come with a separate PR.

File: examples/function-templates/javascript/contentful-app-manifest.json (hunk 1 above)
Comment (cemreyuksel) [reply examples/function-templates/javascript/contentful-app-manifest.json:9]: This just for renaming. That change will come with a separate PR.

Comment (ghepting) [reply examples/function-templates/javascript/contentful-app-manifest.json:9]: ah k. thx :)

**Review by ghepting — APPROVED** [review]

File: examples/function-templates/javascript/INSTRUCTIONS.md [hunk 3]
```diff
@@ -13,9 +13,9 @@ You will need to set the following environment variables as listed below:
 It as simple using the CLI command `npm run upload-ci`. This will perform two actions: upload the code, linking it to the app, and then finally activating the code ready for usage in both 
 
 ### Assigning an app to a field
-Delivery functions are meant to help with resolving field data, meaning the app has to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). 
+Functions are meant to help with resolving field data, meaning the app has to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/).
```
Comment (ghepting) [inline examples/function-templates/javascript/INSTRUCTIONS.md:16]: i don't understand these DF readme instruction changes... you HAVE to be specific about specific function context instructions at some point and renaming all of this here doesn't even make sense to me... ?

Comment (ghepting) [reply examples/function-templates/javascript/INSTRUCTIONS.md:16]:
```suggestion
Functions that support resolving field data on the GraphQL API have to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). 
```

Comment (ghepting) [reply examples/function-templates/javascript/INSTRUCTIONS.md:16]: Perhaps this suggestion helps clarify my concerns/how we should frame this to developers?

File: examples/function-templates/javascript/INSTRUCTIONS.md [hunk 4]
```diff
@@ -13,9 +13,9 @@ You will need to set the following environment variables as listed below:
 It as simple using the CLI command `npm run upload-ci`. This will perform two actions: upload the code, linking it to the app, and then finally activating the code ready for usage in both 
 
 ### Assigning an app to a field
-Delivery functions are meant to help with resolving field data, meaning the app has to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). 
+Functions are meant to help with resolving field data, meaning the app has to be assigned to a field location. You can read more about different app locations [here](https://www.contentful.com/developers/docs/extensibility/app-framework/locations/). 
 
-The field is required to be one of the supported types for delivery function: Short text (`Symbol`) or JSON fields (`Object`).
+The field is required to be one of the supported types for function: Short text (`Symbol`) or JSON fields (`Object`).
```
Comment (ghepting) [inline examples/function-templates/javascript/INSTRUCTIONS.md:18]:
```suggestion
Supported field types for functions are: Short text (`Symbol`) or JSON fields (`Object`).
```

File: examples/function-templates/javascript/INSTRUCTIONS.md (hunk 3 above)
Comment (cemreyuksel) [reply examples/function-templates/javascript/INSTRUCTIONS.md:16]: perfect, thanks!

**Comment by cemreyuksel** [comment]
@BasKiers changes examples as well, according to your suggestion.

**Review by BasKiers — APPROVED** [review]

**Comment by pciazynski** [comment]
I think `"exclude": ["delivery-functions"]` should be updated as well in `examples/function-potterdb/tsconfig.json` and `examples/function-potterdb-rest-api/tsconfig.json`.

**Pull request merged**